src/app/dashboard/components/dashboard-disease-menu/dashboard-disease-menu.component.ts

```diff
@@ -15,8 +16,56 @@ export class DashboardDiseaseMenuComponent implements OnInit, OnDestroy {
   currentFilters: Filter[] = [];
   diseases: any[] = [];
   subscriptions: Subscription[] = [];
+  intervalTimer: any;
+  isIntervalOn: boolean = false;
+  selectedIndex: number = 0;
+
+  progressbarValue = 100;
+  curSec: number = 0;
+  duration: number = 5;
+  colorMap: any = DISEASES_COLORS_MAP;
+
+  progressBarColor: string;
+  defaultProgressBarColor: string = '#515B6B';
+
   constructor(public filterService: FilterService, public enumToKeyValuePipe: EnumToKeyValuePipe) {
     this.diseases = enumToKeyValuePipe.transform(Disease);
+
+    if (this.diseases.length) {
+      const newColor = this.colorMap[this.diseases[0].key];
+      if (newColor) {
```

Comment:
This can be simplified I think:  this.progressBarColor = newColor || this.defaultProgressBarColor  or something similar.